Sources/ODP/OdpConfig.swift

```diff
@@ -0,0 +1,145 @@
+    private var _odpServiceIntegrated: OdpConfigState
+
+    enum OdpConfigState {
+        case notDetermined
+        case integrated
+        case notIntegrated
+    }
+    
+    let queue = DispatchQueue(label: "odpConfig")
+    
+    init(apiKey: String? = nil, apiHost: String? = nil, segmentsToCheck: [String] = []) {
+        self._apiKey = apiKey
+        self._apiHost = apiHost
+        self._segmentsToCheck = segmentsToCheck
+        self._odpServiceIntegrated = .notDetermined  // initially queueing allowed until the first datafile is parsed
+    }
+
+    func update(apiKey: String?, apiHost: String?, segmentsToCheck: [String]) -> Bool {
+        if (apiKey != nil) && (apiHost != nil) {
+            self.odpServiceIntegrated = .integrated
+        } else {
+            // disable future event queueing if datafile has no ODP integrations.
+            self.odpServiceIntegrated = .notIntegrated
+        }
+
+        if self.apiKey == apiKey, self.apiHost == apiHost, self.segmentsToCheck == segmentsToCheck {
+            return false
+        } else {
+            self.apiKey = apiKey
+            self.apiHost = apiHost
+            self.segmentsToCheck = segmentsToCheck
+            return true
+        }
+    }
+}
+
+// MARK: - Thread-safe
+
+extension OdpConfig {
+    
+    var apiHost: String? {
+        get {
+            var value: String?
+            queue.sync {
+                value = _apiHost
+            }
+            return value
+        }
+        set {
+            queue.async {
+                self._apiHost = newValue
+            }
+        }
+    }
+
+    var apiKey: String? {
+        get {
+            var value: String?
+            queue.sync {
+                value = _apiKey
+            }
+            return value
+        }
+        set {
+            queue.async {
+                self._apiKey = newValue
+            }
+        }
+    }
+    
+    var segmentsToCheck: [String] {
+        get {
+            var value = [String]()
+            queue.sync {
+                value = _segmentsToCheck
+            }
+            return value
+        }
+        set {
+            queue.async {
+                self._segmentsToCheck = newValue
+            }
+        }
+    }
+    
+    var odpServiceIntegrated: OdpConfigState {
+        get {
+            var value = OdpConfigState.notDetermined
+            queue.sync {
+                value = _odpServiceIntegrated
+            }
+            return value
+        }
+        set {
+            queue.async {
+                self._odpServiceIntegrated = newValue
+            }
+        }
+    }
+    
+    var eventQueueingAllowed: Bool {
+        var value = true
+        queue.sync {
+            switch _odpServiceIntegrated {
+            case .notDetermined, .integrated: value = true
+            case .notIntegrated: value = false
+            }
+        }
+        return value
+    }
+    
+    var isOdpIntegrated: Bool {
```

Comment:
can we have a test to assert this?

Reply:
I see this is not used any more. It's removed.